Tests/ApolloCodegenTests/CodeGeneration/Templates/SelectionSet/SelectionSetTemplate_Initializers_Tests.swift

```diff
@@ -0,0 +1,758 @@
+          ]))
+        }
+      """
+    
+    // when
+    try buildSubjectAndOperation()
+    
+    let allAnimals = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    )
+    
+    let actual = subject.render(field: allAnimals)
+    
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 61, ignoringExtraLines: true))
+  }
+
+  func test__render_given_fieldWithAlias_rendersInitializer() throws {
+    // given
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    type Animal {
+      species: String!
+    }
+    """
+
+    document = """
+    query TestOperation {
+      allAnimals {
+        aliased: species
+      }
+    }
+    """
+
+    let expected = """
+        public init(
+          aliased: String
+        ) {
+          let objectType = TestSchema.Objects.Animal
+          self.init(data: DataDict(
+            objectType: objectType,
+            data: [
+              "__typename": objectType.typename,
+              "aliased": aliased
+          ]))
+        }
+      """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let allAnimals = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    )
+
+    let actual = subject.render(field: allAnimals)
+
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 15, ignoringExtraLines: true))
+  }
+  
+  func test__render_given_entityFieldSelection_rendersInitializer() throws {
+    // given
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    type Animal {
+      species: String!
+      friend: Animal!
+    }
+    """
+
+    document = """
+    query TestOperation {
+      allAnimals {
+        friend {
+          species
+        }
+      }
+    }
+    """
+
+    let expected = """
+        public init(
+          friend: Friend
+        ) {
+          let objectType = TestSchema.Objects.Animal
+          self.init(data: DataDict(
+            objectType: objectType,
+            data: [
+              "__typename": objectType.typename,
+              "friend": friend._fieldData
+          ]))
+        }
+      """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let allAnimals = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    )
+
+    let actual = subject.render(field: allAnimals)
+
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 15, ignoringExtraLines: true))
+  }
+
+  func test__render_given_abstractEntityFieldSelectionWithNoFields_rendersInitializer() throws {
+    // given
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    interface Animal {
+      species: String!
+      friend: Animal!
+    }
+
+    type Cat implements Animal {
+      species: String!
+      friend: Animal!
+    }
+    """
+
+    document = """
+    query TestOperation {
+      allAnimals {
+        ... on Cat {
```

Comment:
How would the fields in this selection be initialized?

Reply:
The generated `Data.AllAnimal.AsCat` object has it's own initializer that can be used for this!